### PR TITLE
MF-875 - Add tracing to official documentation

### LINKED ISSUE #875: Add Open Tracing to official documentation
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.
No

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
Add a chapter to the official documentation to show developers how to use Jaeger and OpenTracing with Mainflux.

This should include:

- A short explanation of Open Tracing and Jaeger
- How to launch and access the Jaeger UI from Mainflux
- How to configure Jaeger (enable/disable, set ports)
- A small example of _things_ and using Jaeger to find where they can be made more efficient.

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
should-have

I plan to write this documentation myself.  As mentioned by @drasko in #787 , I am also working a blog post showing how to use Jaeger in Mainflux.  I plan to go over the Jaeger UI in this post but I don't believe it should be included in the Mainflux documentation.  The UI is covered by the Jaeger documentation and I think it would be extra work to keep a version in sync in the Mainflux documentation.

### PR DESCRIPTION
**Pull request by nwneisen**
### What does this do?
Adds a chapter to the official documentation to inform users about tracing and Jaeger.

### Which issue(s) does this PR fix/relate to?
Resolves #875 

### List any changes that modify/break current functionality
None

### Have you included tests for your changes?
NA

### Did you document any new/modified functionality?
Yes

### Notes
Uses `mainflux-cli provision test` to create traces that can be viewed as an example.

Added a Configure section which mentions a way to enable/disable the Jaeger service and change the ports used by Jaeger.  If these are not the recommended methods for doing these then they should probably be removed.

### CODE REVIEW

**Review by nmarcetic — DISMISSED** [review]

File: docs/tracing.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Tracing
+
+Distributed tracing is a method of profiling and monitoring applications.  It can provide valuable insight when optimizing and debugging an application.  Mainflux includes the Jaeger open tracing framework as a service with its stack by default.
```
Comment (nmarcetic) [inline docs/tracing.md]: Can you please just link a Jeager to official webister ?

Comment (nwneisen) [reply docs/tracing.md]: Good idea.

File: docs/tracing.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Launch
+
+The Jaeger service will launch with the rest of the Mainflux services.  All services can be launched using:
+
+```bash
+make run
+```
+
+The Jaeger UI can then be accessed at ```http://http://localhost:16686``` from a browser.  Details about the UI can be found in [Jaeger's official documentation](https://www.jaegertracing.io/docs/1.14/frontend-ui/).
```
Comment (anovakovic01) [inline docs/tracing.md]: Remove first `http://`.

**Review by anovakovic01 — CHANGES_REQUESTED** [review]

File: docs/tracing.md (hunk 2 above)
Comment (nwneisen) [reply docs/tracing.md]: Oops! Thanks!

**Review by anovakovic01 — APPROVED** [review]
LGTM!

**Comment by drasko** [comment]
@nwneisen this looks good, please update the branch so we can merge.

**Review by nmarcetic — APPROVED** [review]

**Comment by nwneisen** [comment]
@drasko branch updated.

**Comment by drasko** [comment]
Merged!

**Pull request merged**